### ISSUE TITLE
Group elements unnormalised after repeated application of composition.

### ISSUE THREAD

**Issue by rmurai0610** [facebookresearch/theseus#485]
## 🐛 Bug

Lie group elements are not normalized after each operation, and hence fails _check_tensor test after many repeated application of operation such as composition.

## Steps to Reproduce

```
import theseus as th
import torch

torch.manual_seed(0)

x = th.SE2(x_y_theta=torch.randn(1, 3))

for i in range(100000):
    m = th.SE2(x_y_theta=torch.randn(1, 3))
    x = x.compose(m)

```
## Expected behavior
By normalizing after every operations, the lie group element are always normalize as expected.
```
import theseus as th
import torch

torch.manual_seed(0)

x = th.SE2(x_y_theta=torch.randn(1, 3))

for i in range(100000):
    m = th.SE2(x_y_theta=torch.randn(1, 3))
    x = x.compose(m)
    x = th.SE2(tensor=th.SE2.normalize(x.tensor))

```
## System Info

- OS (e.g., Linux): Ubuntu 20.04
- Python version: 3.8.13
- GPU models and configuration: 3080 Ti
- CUDA version: 11.7
- pip/conda dependencies packages versions: theseus-ai 0.1.4
- Any other relevant information:
